wazo_plugind/http.py

```diff
@@ -34,6 +35,9 @@
 auth_verifier = AuthVerifier()
 
 
+_status_aggregator = None
```

Comment:
Well because when I pass the `status_aggregator` variable that I initiated in `controller.py` like the following:
```python
flask_app = http.new_app(
            config,
            plugin_service=plugin_service,
            status_aggregator=self._status_aggregator,
        )
```
Then we have to add the `StatusChecker` resource (`http.py`):
```python
MultiAPI(APIv02).add_resource(StatusChecker)
```
when we do this above, I did not find a way to pass the `status_aggregator`as a parameter (I do not even know if it is possible; like maybe change the implementation of `add_resource` function?)

The other reason for using global variables; and probably the main one is that in this code:
```python
        flask_app = http.new_app(
            config,
            plugin_service=plugin_service,
            status_aggregator=self._status_aggregator,
        )
        if self._ssl_cert_file and ssl_key_file:
            logger.warning(
                'Using service SSL configuration is deprecated. Please use NGINX instead.'
            )
            wsgi.WSGIServer.ssl_adapter = http_helpers.ssl_adapter(
                self._ssl_cert_file, ssl_key_file
            )
        wsgi_app = ReverseProxied(
            ProxyFix(wsgi.WSGIPathInfoDispatcher({'/': flask_app}))
        )
        self._server = wsgi.WSGIServer(bind_addr=bind_addr, wsgi_app=wsgi_app)
        for route in http_helpers.list_routes(flask_app):
            logger.debug(route)

        if not flask_app.config['auth'].get('master_tenant_uuid'):
            self._token_renewer.subscribe_to_next_token_details_change(
                http.master_tenant.init_value
            )
        self._token_renewer.subscribe_to_next_token_details_change(
            lambda t: self._token_renewer.emit_stop()
        )
        self._token_renewer.subscribe_to_token_change(
            self._token_status.token_change_callback
        )
        self._status_aggregator.add_provider(self._token_status.provide_status)
        http._update_status_aggregator(self._status_aggregator)
```

After we initiate `flask_app` object and pass it the `status_aggregator`; at that precise moment we still did not provide/update the current status of `service_token`, and the `master_tenant` is initiated after `flask_app` initiation, and also it is not possible to move that code (`flask_app` initiation after these two). So the solution that I made was to update the status aggregator in two stages: the first one where we pass the object during the app creation, and then after we initiate the `master_tenant` and we renew the token, we update it again using the function `_update_status_aggregator`